### PR TITLE
Add customize precompile

### PR DESCRIPTION
**Pull request by Jason-Wanxt**

### CODE REVIEW

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:3]:
```suggestion
sidebar_label: "Customize your chain's precompile"
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:17]:
```suggestion
There are 2 ways to customize your own precompile:

 1. Add your method logic to an existing [precompile](https://github.com/OffchainLabs/nitro-contracts/tree/97cfbe00ff0eea4d7f5f5f3afb01598c19ddabc4/src/precompiles).
 2. Create a new precompile.
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:19]:
```suggestion
### Prerequisites
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:65]:
```suggestion
In the above example, `0x0c49c36c` is the hex-encoded, utf8 representation of `hi`.
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:67]:
```suggestion
### Call your function using Foundry's `cast call`
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:73]:
```suggestion
You should see something like this:
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:81]:
```suggestion
First, navigate to the [`precompiles/` directory](https://github.com/OffchainLabs/nitro/tree/master/precompiles) and create a new precompile 
called `ArbSys.sol`.  We'll define a new method, and we'll give it an address:
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:88]:
```suggestion
	Address addr // 0x11a, for example
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:96]:
```suggestion
Then, update [precompile.go](https://github.com/OffchainLabs/nitro/blob/master/precompiles/precompile.go) to register the new precompile under the `Precompiles()` method:
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:99]:
```suggestion
insert(MakePrecompile(templates.ArbHiMetaData, &ArbHi{Address: hex("11a")})) // 0x011a here is an example address
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:102]:
```suggestion
Go to [contracts/src/precompiles/](https://github.com/OffchainLabs/nitro-contracts/tree/97cfbe00ff0eea4d7f5f5f3afb01598c19ddabc4/src/precompiles) and create `ArbHi.sol`. We'll specify the corresponding interface here. Ensure that the method name on the interface matches the name of the function you introduced in the previous step, `camelCased`:
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:115]:
```suggestion
Next, build Nitro by following the instructions in [How to build Nitro locally](/node-running/how-tos/build-nitro-locally). Note that if you've already built the Docker image, you still need run the last step to rebuild.

Run Nitro by following the instructions in [How to run a full node](/node-running/how-tos/running-a-full-node#putting-it-all-together).
```

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:117]:
```suggestion
Once your node is running, you can call `ArbSys.sol` either directly using `curl`, or through Foundry's `cast call`.
```

How do you feel about adding a "Call your precompile" H2 section as the last section, below the two "Option" H2s to minimize duplication?

**Review by symbolpunk — CHANGES_REQUESTED** [review]
This is fantastic. Some minor editorial nits and one question :)

**Comment by symbolpunk** [comment]
[+ @mahsamoosavi @TucksonDev for a spot-check on the precision of my suggested edits]

**Review by mahsamoosavi — CHANGES_REQUESTED** [review]

**Comment by mahsamoosavi** [comment]
> [+ @mahsamoosavi @TucksonDev for a spot-check on the precision of my suggested edits 🙏 ]

Thanks @Jason-Wanxt and @symbolpunk. All those reviews are awesome, I just left very few ones.

**Comment by Jason-Wanxt** [comment]
Thanks @symbolpunk and @mahsamoosavi for your review!

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:113]:
```suggestion
/// Precompiled contract that exists in every Arbitrum chain at 0x000000000000000000000000000000000000011a.
```

Should this be updated to clarify that it's a custom precompiled contract?

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (Jason-Wanxt) [reply arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md]: Sure, we can change it to the sentence you mentioned

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (TucksonDev) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:27]: Since this is related to Orbit chains, it might be good to download the latest orbit-ready nitro branch/tag. The parameter for doing that is --branch, or -b, iirc.

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (TucksonDev) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:34]:
```suggestion
Using your favorite code editor, open the implementation file of an existing precompile from the [precompiles/](https://github.com/OffchainLabs/nitro/tree/master/precompiles) directory. We'll use `ArbSys.sol` as an example. Open the the corresponding `go` file (`ArbSys.go`) and add a simple `SayHi` method:
```

In the precompiles reference page, we talk about the "go implementation file" and the "solidity interface file" to make a distinction between both assets, so it is more clear for the user. I've added one potential example if you want to use the same language.

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (TucksonDev) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:42]:
```suggestion
Then, open the corresponding Solidity interface file (`ArbSys.sol`) from the [contracts/src/precompiles/](https://github.com/OffchainLabs/nitro-contracts/tree/97cfbe00ff0eea4d7f5f5f3afb01598c19ddabc4/src/precompiles) directory and add the required interface. Ensure that the method name on the interface matches the name of the function you introduced in the previous step, `camelCased`:
```

(Following latest comment)

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (TucksonDev) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:48]:
```suggestion
Next, build Nitro by following steps 3-7 of the instructions in [How to build Nitro locally](/node-running/how-tos/build-nitro-locally). Note that if you've already built the Docker image, you still need run the last step to rebuild.
```

Adding the steps just in case, to prevent them for cloning nitro again .

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (TucksonDev) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md:50]: Idea: as this is specific for orbit chains, we could maybe add the command here and a note below saying something like "to know more about all available parameters read [How to run a full node](/node-running/how-tos/running-a-full-node#putting-it-all-together)".

**Comment by TucksonDev** [comment]
This looks amazing, Jason, thanks! 🚀 
Added a few more comments for potential minor and QoL improvements, but there's nothing blocking from my side. I'll let @mahsamoosavi and @symbolpunk approve it as they requested some changes :)

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (Jason-Wanxt) [reply arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md]: Yes, we need other command line to run orbit, how do you think if we point them to the orbit setup script then use that docker-compose file and tell user to change the image name.

Comment (TucksonDev) [reply arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md]: Oh, that's right. That's better, yeah. Show them where to change the image name in the docker-compose file.

Comment (symbolpunk) [reply arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md]: Love it, definitely aligned with the general principle of "avoid forcing readers to go over there and come back".

I probably wouldn't block this PR on its implementation, but we can optimize via another PR, eg by partializing the command and including in both places.

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.mdx
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.mdx:69]:
```suggestion
`0x6869` is the hex-encoded utf8 representation of `hi`
```

Calling @Jason-Wanxt @TucksonDev for a sanity check on this edit 🙏 

Future optimization -> It's not clear to me what this is referring to - I'd expect to see `0x6869` in the above sample output. I do see `...00000002686900000000...`.

File: arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md
Comment (symbolpunk) [reply arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md]: Love this - thank you

**Review by mahsamoosavi — APPROVED** [review]
LGTM!

**Review by symbolpunk — APPROVED** [review]

**Pull request merged**
